api4/channel.go

```diff
@@ -99,7 +114,28 @@ func createChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	sc, appErr := c.App.CreateChannelWithUser(c.AppContext, channel, c.AppContext.Session().UserId)
+	var category *model.SidebarCategoryWithChannels
+	categoryId := data.CategoryId
+	if categoryId != "" {
+		if !model.IsValidId(categoryId) {
+			c.SetInvalidParam("category_id")
+			return
+		}
+		if len(categoryId) == 26 {
```

Comment:
I believe this check is redundant. `model.IsValidId` already checks if length == 26.